src/goto-programs/show_goto_functions_json.cpp

```diff
@@ -41,10 +41,13 @@ json_objectt show_goto_functions_jsont::convert(
 {
   json_arrayt json_functions;
   const json_irept no_comments_irep_converter(false);
-  for(const auto &function_entry : goto_functions.function_map)
+
+  const auto sorted = goto_functions.sorted();
+
+  for(const auto function_entry : sorted)
```

Comment:
Minor note: here (and also in the XML case) this should likely be a `const auto &`. It's just an iterator so not terribly expensive to copy, but it's being done in the `PLAIN` case and would thus at bare minimum ensure consistency.

Reply:
I'd be tempted to remove the & in the `PLAIN` case. References to iterators are unusual; copying them is very common.

Reply:
I have no strong opinion either way, just consistency is what I'd call for.